exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -792,124 +1286,344 @@ func (e *Exchange) GetFeeByType(ctx context.Context, feeBuilder *exchange.FeeBui
 
 // GetActiveOrders retrieves any orders that are active/open
 func (e *Exchange) GetActiveOrders(ctx context.Context, req *order.MultiOrderRequest) (order.FilteredOrders, error) {
-	err := req.Validate()
-	if err != nil {
-		return nil, err
-	}
-
-	resp, err := e.GetOpenOrdersForAllCurrencies(ctx)
-	if err != nil {
-		return nil, err
+	var samplePair currency.Pair
+	if len(req.Pairs) == 1 {
+		samplePair = req.Pairs[0]
 	}
-
-	format, err := e.GetPairFormat(asset.Spot, false)
-	if err != nil {
-		return nil, err
+	var sideString string
+	switch {
+	case req.Side.IsLong():
+		sideString = order.Buy.String()
+	case req.Side.IsShort():
+		sideString = order.Sell.String()
 	}
-
 	var orders []order.Detail
-	for key := range resp.Data {
-		var symbol currency.Pair
-		symbol, err = currency.NewPairDelimiter(key, format.Delimiter)
+	switch req.AssetType {
+	case asset.Spot:
+		resp, err := e.GetOpenOrders(ctx, samplePair, sideString, "", req.FromOrderID, 0)
 		if err != nil {
 			return nil, err
 		}
-		for i := range resp.Data[key] {
+		for _, td := range resp {
+			cp, err := currency.NewPairFromString(td.Symbol)
+			if err != nil {
+				return nil, err
+			}
+			if len(req.Pairs) != 0 && req.Pairs.Contains(cp, true) {
```

Comment:
Are you sure you wish to skip the entry if it is a pair that is contained in requested pairs?